kotlin/internal/jvm/android.bzl

```diff
@@ -63,19 +64,19 @@ def _kt_android_artifact(
             **kwargs
         )
         exported_target_labels.append(base_name)
-    elif resource_files or manifest:
+    elif resource_files or manifest or assets:
         # TODO(https://github.com/bazelbuild/rules_kotlin/issues/556): replace with starlark
         # buildifier: disable=native-android
         # Do not export deps to avoid all upstream targets to be invalidated when ABI changes.
+        kwargs.pop("exports", default = None)
         _android_library(
             name = base_name,
             resource_files = resource_files,
             deps = deps,
-            custom_package = kwargs.get("custom_package", default = None),
-            manifest = manifest,
             enable_data_binding = enable_data_binding,
             tags = tags,
             visibility = ["//visibility:private"],
+            **kwargs
```

Comment:
I'd still just pass the assets here instead of the entire `kwargs`.

Reply:
It's not just assets though. If assets is provided, assets_dir must be provided. Providing one and not the other causes an error in android_library. This is a prime example of why we shouldn't be white-listing which arguments to pass down. 

As another example, we have libraries that use manifest_values and some using proguard_specs. Checking for and whitelisting individual attributes is not a reliable path forward.

Reply:
@Bencodes : yes, this is to avoid exports which causes extra invalidation / increased build time. But we ran into 2 issues since then 1) assets are not handled indeed 2) resource-less module don't have their manifest merged upstream. I can share the update we've made to fix this if this helps (and we didn't get any issues since then).

Reply:
here we go https://github.com/bazelbuild/rules_kotlin/pull/1113